File: algorithms/selection_sort.php
<?php
    include_once "library.php"; // The programe executaion will continue
                                // even if the "library.php" wasn't included
    require_once "library.php"; // In case of requiring the "library.php" file
                                // the programe will not execute if there is 
                                // an error in importing the file

    function SSort(& $list)
    {
        for ($i=0; $i < sizeof($list) ; $i++) { 
            $minValue = $list[$i];
            $minIndex = $i;
            for ($j=$i; $j < sizeof($list) ; $j++) { 
                if ($list[$j] < $minValue ) {
                    $minValue = $list[$j];
                    $minIndex = $j;
                }
            }
            if ($i !== $minIndex ) {
                replace($list[$minIndex], $list[$i]);
            }
        }
    }

    $list = array( 3, 2, 6, 5, 1, 4 );
    SSort($list);
    printArray($list);
?>

File: algorithms/selection_sort.cpp
#include "library.h"

void sort(int n, int* list)
{
    int min_value, min_index;
    for(int i = 0; i<n; i++){
        min_value = list[i];
        min_index = i;
        for(int j = i; j<n; j++){
            if(list[j]<min_value){
                min_value = list[j];
                min_index = j;
            }
        }
        if(min_index != i){
            replace(list[i], list[min_index]);
        }
    }
}

int main()
{
    int arr[] = {9, 5, 7, 3, 1};
    sort(5, arr);
    print(5, arr);
}